instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2.rb

```diff
@@ -11,6 +11,45 @@ module OpenTelemetry
   module Instrumentation
     # Contains the OpenTelemetry instrumentation for the Mysql2 gem
     module Mysql2
+      extend self
+
+      CURRENT_ATTRIBUTES_KEY = Context.create_key('mysql-attributes-hash')
```

Comment:
I wonder if we should change the redis context key to be more specific as well. 

This is what context storage looks like in a block like this
```rb
OpenTelemetry::Instrumentation::Redis.with_attributes({ 'foo' => 'bar' }) do
  OpenTelemetry::Common::HTTP::ClientContext.with_attributes({'bar' => 'foo'}) do
    binding.pry
  end
end
```

```rb
=> [#<OpenTelemetry::Context:0x00000001071447b0
  @entries=
   {#<OpenTelemetry::Context::Key:0x000000012729d448
     @name="current-attributes-hash">=>{"foo"=>"bar"}}>,
 #<OpenTelemetry::Context:0x0000000107144698
  @entries=
   {#<OpenTelemetry::Context::Key:0x000000012729d448
     @name="current-attributes-hash">=>{"foo"=>"bar"},
    #<OpenTelemetry::Context::Key:0x000000012711f6e8
     @name="current-attributes-hash">=>{"bar"=>"foo"}}>]
```

I've never actually had to debug this ever, but it might be a kindness to future us if we ever find ourselves in that predicament. 

(I don't expect that change to be included in this PR just wanted to raise that thought here where its relevant.)

Reply:
Yeah, I think this is a great point! Will address in separate PR.

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/pull/1186